### PR TITLE
BL-788 Print dialog appears behind Bloom (Linux)

### PR DESCRIPTION
**Pull request by davidmoore1**
Added small pauses after minimizing and before restoring to
eliminate errors that were observed after the first fix.  Also
added changes to disable buttons in the publish view until
the print operation is actually complete.

### CODE REVIEW

**Comment by sillsdevgerrit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.lsdev.sil.org//job/GitHub-Bloom-Wrapper-master-debug/418/
Test FAILed.

**Comment by davidmoore1** [comment]
Retest this please

**Comment by sillsdevgerrit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.lsdev.sil.org//job/GitHub-Bloom-Wrapper-master-debug/419/
Test FAILed.

**Comment by davidmoore1** [comment]
Retest this please

**Comment by davidmoore1** [comment]
Retest this please

**Comment by sillsdevgerrit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.lsdev.sil.org//job/GitHub-Bloom-Wrapper-master-debug/420/
Test FAILed.

**Comment by davidmoore1** [comment]
Retest this please

**Comment by sillsdevgerrit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.lsdev.sil.org//job/GitHub-Bloom-Wrapper-master-debug/423/
Test FAILed.

**Comment by davidmoore1** [comment]
Retest this please

File: src/BloomExe/Publish/PdfViewer.cs [hunk 1]
```diff
@@ -98,21 +105,106 @@ public void Print()
...
+					print.Print (null, this);
+				} catch (COMException e) {
+					if (PrintProgress != null) {
+						PrintProgress.Invoke (this, new PdfPrintProgressEventArgs(false));
+					}
+					//NS_ERROR_ABORT means user cancelled the printing, not really an error.
+					if (e.ErrorCode != GeckoError.NS_ERROR_ABORT)
+						Console.WriteLine ("ComError");
+				}
+				Marshal.ReleaseComObject (print);
```
Comment (StephenMcConnel) [inline src/BloomExe/Publish/PdfViewer.cs]: should this Marshal.ReleaseComObject() be inside a finally {} block?

File: src/BloomExe/Publish/PublishView.cs [hunk 2]
```diff
@@ -466,7 +484,21 @@ private void OnPrint_Click(object sender, EventArgs e)
...
 			Analytics.Track("Print PDF");
 		}
+		private void OnPrintProgress(object sender, PdfPrintProgressEventArgs e)
+		{
+			// BL-788 Only called in Linux version.  Protects against button
+			// pushes while print is in progress.
+			if (e.PrintInProgress)
+			{
+				SetDisplayMode(PublishModel.DisplayModes.Printing);
+			}
```
Comment (StephenMcConnel) [inline src/BloomExe/Publish/PublishView.cs]: Do you need the UpdateDisplay() in both branches of the if, or only in the else branch as it is now?

**Comment by StephenMcConnel** [comment]
Generally looks good. I do have a couple of questions, the first more serious than the second.

**Comment by StephenMcConnel** [comment]
:+1:

**Comment by davidmoore1** [comment]
Made a change, prior to Steve's final review addressing the first issue.  The answer to the second is, no you don't want it both places.  In the other case, UpdateDisplay would reenable the buttons that we are disabling in the set display mode.  In the other case, after the print, it restores the buttons to the state they should be in, which enables the outside cover and inside pages buttons only if the underlying model says they should.

**Pull request merged**